nilearn/image/image.py

```diff
@@ -205,18 +208,25 @@ def _smooth_array(arr, affine, fwhm=None, ensure_finite=True, copy=True):
         # SPM tends to put NaNs in the data outside the brain
         arr[np.logical_not(np.isfinite(arr))] = 0
 
-    if fwhm == 'fast':
+    if isinstance(fwhm, str) and (fwhm == 'fast'):
```

Comment:
We don't need `isinstance(fwhm, str)` if it's in `and` with an equivalence test with a string.

Reply:
we do, in case fwhm is a numpy array

Reply:
if `fwhm` is a numpy array, the if statement will fail the `fwhm=='fast'` comparison regardless.

Reply:
I'm guessing its the same thing here, `fwhm=='fast'` will do an element-wise comparision, which is why the `isinstance` coparision is necessary?

Reply:
Would `fwmh is 'fast'` work here instead of thse 2 comparisions? 
Regardless, I always forget this one, let's add a comment here explaining this for future code perusers.

Reply:
in some cases strings can be equal but not have the same id:

https://stackoverflow.com/questions/1504717/why-does-comparing-strings-using-either-or-is-sometimes-produce-a-differe

so @pausz 's solution is the correct one here

Reply:
> I'm guessing its the same thing here, `fwhm=='fast'` will do an element-wise comparision, which is why the `isinstance` comparison is necessary?

exactly. `fwhm == 'fast'` only gets evaluated if the first condition is true